ssv/runner.go

```diff
@@ -234,6 +212,11 @@ func (b *BaseRunner) hasRunningDuty() bool {
 }
 
 func (b *BaseRunner) ShouldProcessDuty(duty *types.Duty) error {
+	// first height special case
```

Comment:
*Why do we need it?*

It is a special case because of an implementation detail. Currently we bootstrap the Controller's state with 0. Also the first duty can also be decided with height 0. In all other cases the duty will be decided on a height greater than the controller.

We can do a smart fix in spec so that 0 height will work like any other height and won't need special conditions in our algos (not in this PR).

However, even if we do that maybe we should have first height tests. We can't do check for all heights but first height is suspect for bugs.

*do we test it?*
See `valid_first_height.go`